dojo/importers/base_importer.py

```diff
@@ -237,58 +192,62 @@ def parse_dynamic_test_type_findings_from_tests(
 
     def parse_findings_dynamic_test_type(
         self,
-        parser: Parser,
-        scan_type: str,
         scan: TemporaryUploadedFile,
-        **kwargs: dict,
+        parser: Parser,
     ) -> List[Finding]:
         """
         Use the API configuration object to get the tests to be used by the parser
         to dump findings into
 
         This version of this function is intended to be extended by children classes
         """
-        tests = self.parse_dynamic_test_type_tests(
-            parser,
-            scan_type,
-            scan,
-            **kwargs,
-        )
-        return self.parse_dynamic_test_type_findings_from_tests(tests, **kwargs)
+        tests = self.parse_dynamic_test_type_tests(scan, parser)
+        return self.parse_dynamic_test_type_findings_from_tests(tests)
 
     def parse_findings(
         self,
-        parser: Parser,
-        scan_type: str,
         scan: TemporaryUploadedFile,
-        test: Test = None,
-        **kwargs: dict,
+        parser: Parser,
     ) -> List[Finding]:
         """
         Determine how to parse the findings based on the presence of the
         `get_tests` function on the parser object
         """
-        if hasattr(parser, 'get_tests'):
-            return self.parse_findings_dynamic_test_type(
-                parser,
-                scan_type,
-                scan,
-                **kwargs,
-            )
-        else:
-            return self.parse_findings_static_test_type(
-                parser,
-                scan_type,
-                scan,
-                test=test,
-                **kwargs,
-            )
+        # Attempt any preprocessing before generating findings
+        if len(self.parsed_findings) == 0 or self.test is None:
+            scan = self.process_scan_file(scan)
+            if hasattr(parser, 'get_tests'):
+                self.parsed_findings = self.parse_findings_dynamic_test_type(scan, parser)
+            else:
+                self.parsed_findings = self.parse_findings_static_test_type(scan, parser)
+        return self.parsed_findings
+
+    def sync_process_findings(
+        self,
+        parsed_findings: List[Finding],
+        **kwargs: dict,
+    ) -> Tuple[List[Finding], List[Finding], List[Finding], List[Finding]]:
+        """
+        Processes findings in a synchronous manner such that all findings
+        will be processed in a worker/process/thread
+        """
+        return self.process_findings(parsed_findings, sync=True, **kwargs)
+
+    def async_process_findings(
+        self,
+        parsed_findings: List[Finding],
+        **kwargs: dict,
+    ) -> List[Finding]:
+        """
+        Processes findings in chunks within N number of processes. The
+        ASYNC_FINDING_IMPORT_CHUNK_SIZE setting will determine how many
+        findings will be processed in a given worker/process/thread
+        """
+        return self.process_findings(parsed_findings, sync=True, **kwargs)
```

Comment:
Should this be `sync=False`?

```suggestion
        return self.process_findings(parsed_findings, sync=False, **kwargs)
```